api/src/main/java/com/example/api/service/user/DashboardService.java

```diff
@@ -0,0 +1,274 @@
+                .stream()
+                .filter(GraphTaskResult::isEvaluated)
+                .mapToDouble(result -> result.getGraphTask().getMaxPoints())
+                .sum();
+        Double maxPointsFileTask = fileTaskResultRepo.findAllByUser(student)
+                .stream()
+                .filter(FileTaskResult::isEvaluated)
+                .mapToDouble(result -> result.getFileTask().getMaxPoints())
+                .sum();
+        Double maxPointsSurvey = surveyResultRepo.findAllByUser(student)
+                .stream()
+                .filter(SurveyResult::isEvaluated)
+                .mapToDouble(result -> result.getSurvey().getMaxPoints())
+                .sum();
+        return maxPointsGraphTask + maxPointsFileTask + maxPointsSurvey;
+    }
+
+    private List<LastAddedActivity> getLastAddedActivities(User student) {
+        List<GraphTask> graphTasks = graphTaskService.getStudentGraphTasks(student);
+        List<FileTask> fileTasks = fileTaskService.getStudentFileTasks(student);
+        List<Survey> surveys = surveyService.getStudentSurvey(student);
+        List<Info> infos = infoService.getStudentInfos(student);
+
+        return Stream.of(graphTasks, fileTasks, surveys, infos)
+                .flatMap(Collection::stream)
+                .sorted(((o1, o2) -> Long.compare(o2.getCreationTime(), o1.getCreationTime())))
+                .limit(MAX_LAST_ACTIVITIES_IN_DASHBOARD)
+                .map(this::toLastAddedActivity)
+                .toList();
+    }
+
+    private LastAddedActivity toLastAddedActivity(Activity activity) {
+        Chapter chapter = chapterService.getChapterWithActivity(activity);
+        String chapterName = Objects.nonNull(chapter) ? chapter.getName() : null;
+        String activityType = activity.getActivityType().getActivityType();
+        Double points = activity.getMaxPoints();
+        Requirement requirement = activity.getRequirements()
+                .stream()
+                .filter(req -> req.isSelected() && Objects.nonNull(req.getDateFrom()))
+                .findAny()
+                .orElse(null);
+        String availableUntil = Objects.nonNull(requirement) ? new Date(requirement.getDateFrom()).toString() : null;
+        return new LastAddedActivity(chapterName, activityType, points, availableUntil);
+
+    }
+
+    // TODO: Replace mocks
+    private HeroStats getHeroStats(User student) {
+        Double experiencePoints = getExperiencePoints(student);
+        Double nextLvlPoints = 100D; // TODO
+        String rankName = "Nowicjusz"; // TODO
+        Long badgesNumber = 3L; // TODO
+        Long completedActivities = getCompletedActivities(student);
+
+        return new HeroStats(
+                experiencePoints,
+                nextLvlPoints,
+                rankName,
+                badgesNumber,
+                completedActivities
+        );
+    }
+
+    private Double getExperiencePoints(User student) {
```

Comment:
Ogólnie to jest pole w studencie points (to są własnie punkty doświadczenia - zmieniłem, bo i tak wszędzie używamy określenia punkty a nie doświdaczenie), zostało zrobione po to, żeby po każdej ocenie dodawały sie tam punkty - nie wiem czy to jesy robione teraz, jeżeli nie to zamiast tego (bo to się będzie liczyć za każdym razem jak ktos odświeży strone) albo daj taska na Jire żeby punkty się dodawały do pola points po każdej zakończonej aktywności, albo sam to zrób

Reply:
To dodawanie punktów po aktywnościach zrobie w oddzielnym tasku, żeby nie mieszać i wykorzystam to nowe pole zamiast liczenia tego za każdym razem